community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java

```diff
@@ -389,7 +390,19 @@ public void force() throws IOException
         synchronized ( logFile )
         {
             writer.emptyBufferIntoChannelAndClearIt();
+        }
+        // Force the writer outside of the lock.
+        // This allows multiple threads forcing at the same time to piggy-back onto one another.
```

Comment:
@thobe Looking at it a bit more, now that you ask, then perhaps there might actually be a correctness issue... I think that you should do a writer.getChannel() (doesn't exist today) within the synchronized block and then channel.force() on that channel instead of writer.force(), to avoid the risk of forcing on a new channel created after a concurrent rotation.

I also couldn't find any forcing upon rotation, on the old channel/file, which perhaps might be necessary and either way it should be low-impact since it happens infrequently.